agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -74,6 +77,198 @@ namespace beerocks {
+     */
+    sLinkMetrics link_metrics;
+
+    /**
+     * Link metrics collector to use to get link metrics in the connecting interface.
+     */
+    std::unique_ptr<link_metrics_collector> collector;
+
+    /**
+     * Get a link metrics collector suitable for the underlying network technology of the
+     * connecting interface.
+     * Collector choice depends on bits 15 to 8 of media type.
+     */
+    uint8_t media_type_msb = media_type >> 8;
+    if (0 == media_type_msb) {
+        collector = std::make_unique<ieee802_3_link_metrics_collector>();
+    } else if (1 == media_type_msb) {
+        // TODO: Create a collector for wireless interfaces
+        //collector = std::make_unique<ieee802_11_link_metrics_collector>();
+    } else {
+        LOG(ERROR) << "Unsupported media type: " << std::hex << (int)media_type;
+        return (false);
+    }
+
+    /**
+     * Get link metrics information
+     */
+    if (collector &&
+        collector->get_link_metrics(local_interface_name, neighbor_al_mac, link_metrics)) {
+
+        /**
+         * Get local interface information to obtain its MAC address.
+         */
+        network_utils::iface_info iface_info;
+        if (network_utils::get_iface_info(iface_info, local_interface_name) != 0) {
+            LOG(ERROR) << "Failed reading interface info for: " << local_interface_name;
+            return (false);
+        }
+        sMacAddr local_interface_mac = network_utils::mac_from_string(iface_info.mac);
+
+        /**
+         * Add Transmitter Link Metric TLV if specifically requested or both requested
+         */
+        if ((ieee1905_1::eLinkMetricsType::TX_LINK_METRICS_ONLY == link_metrics_type) ||
+            (ieee1905_1::eLinkMetricsType::BOTH_TX_AND_RX_LINK_METRICS == link_metrics_type)) {
+            auto tlvTransmitterLinkMetric =
+                cmdu_tx.addClass<ieee1905_1::tlvTransmitterLinkMetric>();
+            if (!tlvTransmitterLinkMetric) {
+                LOG(ERROR) << "addClass ieee1905_1::tlvTransmitterLinkMetric failed";
+                return false;
+            }
+
+            tlvTransmitterLinkMetric->reporter_al_mac() = reporter_al_mac;
+            tlvTransmitterLinkMetric->neighbor_al_mac() = neighbor_al_mac;
+
+            if (!tlvTransmitterLinkMetric->alloc_interface_pair_info()) {
+                LOG(ERROR) << "alloc_interface_pair_info failed";
+                return false;
+            }
+            auto interface_pair_info = tlvTransmitterLinkMetric->interface_pair_info(0);
+            if (!std::get<0>(interface_pair_info)) {
+                LOG(ERROR) << "Failed accessing interface_pair_info";
+                return false;
+            }
+            auto interfacePairInfo                      = std::get<1>(interface_pair_info);
+            interfacePairInfo.rc_interface_mac          = local_interface_mac;
+            interfacePairInfo.neighbor_interface_mac    = neighbor_al_mac;
```

Comment:
Hm... This is not correct... We actually have to get this from the topology discovery message, which will give us the neighbor interface MAC.

So for now, add a TODO.

Reply:
Done